src/Steam/Provider.php

```diff
@@ -305,6 +311,18 @@ protected function getTokenUrl()
      */
     public static function additionalConfigKeys()
     {
-        return ['realm', 'proxy'];
+        return ['realm', 'proxy', 'allowed_hosts'];
+    }
+
+    /**
+     * Validation of the domain available for authorization.
+     *
+     * @return bool
+     */
+    protected function validateHost(string $url): bool
+    {
+        $allowedHosts = $this->getConfig('allowed_hosts', []);
+
+        return count($allowedHosts) === 0 || in_array(parse_url($url, PHP_URL_HOST), $allowedHosts, true);
```

Comment:
This should be backwards compatible right, as it will return true if no domains are configured?
Maybe add some more comment to readme on why allowed hosts is important?

Reply:
Yes, that is why the setting is ignored if it is missing.
I think it could be added as a description that this setting only allows authorization from certain domains to protect against user authorization spoofing.

Reply:
@atymic Should we add this to the readme?

Reply:
Yes please, not much point in adding this security feature if it defaults to off and isn't documented

Reply:
@atymic Im added this for readme